src/services/asset-discovery-service.ts

```diff
@@ -124,29 +152,40 @@ export default class AssetDiscoveryService extends PercyClientService {
     await this.closeBrowser()
   }
 
-  private async resourcesForWidth(page: puppeteer.Page, width: number, domSnapshot: string,
-                                  rootResourceUrl: string, enableJavaScript: boolean): Promise<any[]> {
+  private async resourcesForWidth(
+    browser: puppeteer.Browser,
+    pooledPage: puppeteer.Page | null,
+    width: number,
+    domSnapshot: string,
+    rootResourceUrl: string,
+    enableJavaScript: boolean,
+  ): Promise<any[]> {
     logger.debug(`discovering assets for width: ${width}`)
-
+  
+    const page = pooledPage || await this.createPage(browser)
```

Comment:
If the `pooledPage` doesn't exist, we create one here. But that newly created page is not added to `this.pages` so I don't think it will ever get cleaned up.

Reply:
Yes, we don't add it to the pool. As it is cleaned up as soon as it is done with: https://github.com/percy/percy-agent/pull/168/files/11b1bd8df53dc0abc0d03bf0dc00fca8999978be#diff-81e4b6d82b5aac3d228d58c084bdc537R225